common/src/main/java/ac/grim/grimac/utils/anticheat/PlayerDataManager.java

```diff
@@ -35,6 +37,11 @@ public GrimPlayer getPlayer(final @NotNull User user) {
         return player;
     }
 
+    @Nullable
+    public User getUser(final @NotNull UUID uuid) {
+        return allUsers.get(uuid);
```

Comment:
the logic copying from https://github.com/GrimAnticheat/Grim/pull/2309/files#r2445597231 can be done like this (plus removing the `allUsers` field and it's usages)
```suggestion
        Object channel = PacketEvents.getAPI().getProtocolManager().getChannel(uuid);
        User user = PacketEvents.getAPI().getProtocolManager().getUser(channel);
        return user;
```